mlflow/models/evaluation/default_evaluator.py

```diff
@@ -41,20 +42,30 @@
 _DEFAULT_SAMPLE_ROWS_FOR_SHAP = 2000
 
 
-def _infer_model_type_by_labels(labels):
-    distinct_labels = set(labels)
-    for v in distinct_labels:
+def _is_categorical(values):
+    distinct_values = set(values)
```

Comment:
Instead of doing manual inspection, can we use Pandas to determine whether it's categorical first? https://pandas.pydata.org/docs/user_guide/categorical.html#categorical-data

Reply:
+1